Packages/Netherlands3D/T3DPipeline/Runtime/Scripts/CityJSONLayer.cs

```diff
@@ -0,0 +1,235 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using Netherlands3D.Core;
+using UnityEngine.Networking;
+using System;
+using System.Linq;
+using UnityEngine.Rendering;
+using Netherlands3D.T3DPipeline;
+using Netherlands3D.Events;
+
+namespace Netherlands3D.TileSystem
+{
+    public class CityJSONLayer : Layer //todo: put in NL3D Package
+    {
+        public ShadowCastingMode tileShadowCastingMode = ShadowCastingMode.On;
+
+        [SerializeField]
+        private GameObject containerPrefab;
+        private GameObject container;
+
+        [SerializeField]
+        private bool checkIfFileExistsBeforeDownloading;
+
+        public override void HandleTile(TileChange tileChange, System.Action<TileChange> callback = null)
+        {
+            TileAction action = tileChange.action;
+            var tileKey = new Vector2Int(tileChange.X, tileChange.Y);
+            switch (action)
+            {
+                case TileAction.Create:
+                    Tile newTile = CreateNewTile(tileKey);
+                    tiles.Add(tileKey, newTile);
+                    break;
+                case TileAction.Upgrade:
+                    tiles[tileKey].unityLOD++;
+                    break;
+                case TileAction.Downgrade:
+                    tiles[tileKey].unityLOD--;
+                    break;
+                case TileAction.Remove:
+                    InteruptRunningProcesses(tileKey);
+                    RemoveGameObjectFromTile(tileKey);
+                    tiles.Remove(tileKey);
+                    callback(tileChange);
+                    return;
+                default:
+                    break;
+            }
+            if (checkIfFileExistsBeforeDownloading)
+                tiles[tileKey].runningCoroutine = StartCoroutine(CheckAndDownloadCityJSON(tileChange, callback));
+            else
+                tiles[tileKey].runningCoroutine = StartCoroutine(DownloadCityJSON(tileChange, callback));
+        }
+
+        private Tile CreateNewTile(Vector2Int tileKey)
+        {
+            Tile tile = new Tile();
+            tile.unityLOD = 0;
+            tile.tileKey = tileKey;
+            tile.layer = transform.gameObject.GetComponent<Layer>();
+            //tile.gameObject = new GameObject();
```

Comment:
I discussed this part with Tom, and the commented code can be removed. I will pick that up a follow-up PR